### PR TITLE
Support for CoreUpdate options

### PR DESCRIPTION
**Pull request by rgbkrk**
This provides the [options for CoreUpdate](https://coreos.com/docs/coreupdate/coreos/coreupdate-configure-machines/).

Follow on PR to coreos-cluster-cli will come after this is merged and released.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Alright, I've gotten it to set the right values

```
core@cluster-1413740766-03 ~ $ cat /etc/coreos/update.conf
GROUP=ccda2076-5b9d-4cd5-b4d4-a14af8ea42ca
REBOOT_STRATEGY=etcd-lock
SERVER=https://jupyter.update.core-os.net/v1/update/
```

and I can see them coming online in the CoreUpdate UI:

screen shot 2014-10-19 at 10 51 09 am

**Comment by rgbkrk** [comment]
This is now ready for review.

File: lib/cloud-config.js [hunk 1]
```diff
@@ -62,6 +62,17 @@ exports.getCloudConfig = function(options, callback) {
 
     template.coreos.etcd.discovery = options.discoveryServiceUrl;
     template.coreos.fleet.metadata = 'provider=rackspace,region=' + options.region;
+    
+    // The update group. Can be "master", "stable", "alpha", or a
+    // UUID for a custom group
+    if(options.update.group) {
```
Comment (kenperkins) [inline lib/cloud-config.js]: Should check for `options.update` before... i.e. `if (options.update && options.update.group) {`. If you don't pre-check `options.update` you could get a `cannot read property server of undefined` error.

Also a style nit: `if (` with a space before the open parenthesis.

File: lib/cloud-config.js [hunk 2]
```diff
@@ -62,6 +62,17 @@ exports.getCloudConfig = function(options, callback) {
...
+        template.coreos.update.group = options.update.group;
+    }
+    
+    // The omaha endpoint URL which will be queried for updates.
+    if(options.update.server) {
```
Comment (kenperkins) [inline lib/cloud-config.js]: Same as above.

File: lib/cluster.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ var Cluster = function(options) {
   this.monitoringToken = options.monitoringToken;
   this.newDiscoveryServiceUrl = options.newDiscoveryServiceUrl || 'https://discovery.etcd.io/new';
   this.privateNetwork = options.privateNetwork;
+  this.update = options.update;
```
Comment (kenperkins) [inline lib/cluster.js]: should this have a reasonable default? i.e. `this.update = options.update || {};`

File: index.js [hunk 4]
```diff
@@ -11,6 +11,8 @@ var Cluster = require('./lib/cluster').Cluster;
  * @param {string}      [options.keyname]     ssh keyname; if not provided will create a new key 'coreos-cluster'
  * @param {string}      [options.privateNetwork]     optional private network guid to use for rackspace private network
  * @param {string}      [options.monitoringToken]    optional monitoring token to configure cloud monitoring
+ * @param {string}      [options.updateGroup]     optional update group (e.g. stable, beta, UUID for a channel)
+ * @param {string}      [options.updateServer]     optional omaha endpoint URL for updates
```
Comment (kenperkins) [inline index.js]: I thought we were going to make update an object here?

File: lib/cloud-config.js (hunk 1 above)
Comment (rgbkrk) [inline lib/cloud-config.js]: Ah, yeah, of course. Thanks.

File: lib/cluster.js (hunk 3 above)
Comment (rgbkrk) [inline lib/cluster.js]: Put that in for now.

We could [configure `reboot-strategy` too](https://coreos.com/docs/cluster-management/setup/update-strategies/) and put in the default, `best-effort`.

```
#cloud-config
coreos:
  update:
    reboot-strategy: best-effort
```

File: index.js (hunk 4 above)
Comment (rgbkrk) [inline index.js]: Yes, yes we are. Check on how I specified this in the latest commit to let me know if its right.

File: index.js [hunk 5]
```diff
@@ -12,7 +12,10 @@ var Cluster = require('./lib/cluster').Cluster;
  * @param {string}      [options.privateNetwork]     optional private network guid to use for rackspace private network
  * @param {string}      [options.monitoringToken]    optional monitoring token to configure cloud monitoring
  * @param {string}      [options.discoverServiceUrl] url for an existing cluster's discover service
- * @param {string}      [options.keyname]     ssh keyname; if not provided will create a new key 'coreos-cluster'
+ * @param {string}      [options.keyname]            ssh keyname; if not provided will create a new key 'coreos-cluster'
+ * @param {object}      [options.update]             options for CoreOS updates
+ * @param {string}      [options.update.group]       optional update group (e.g. stable, beta, UUID for a channel)
+ * @param {string}      [options.update.server]      optional omaha endpoint URL for updates
```
Comment (kenperkins) [inline index.js:18]: Perfecto.

**Comment by kenperkins** [comment]
Looks great!

**Pull request merged**